### PR TITLE
feat: walletd desktop app

### PR DESCRIPTION
**Pull request by alexfreska**
> Note:  Please ignore Windows failures for now.

**Notes**
- walletd binary download flow will not actually work until there is at least one official release.
- This PR depends on YAML support in an official release https://github.com/SiaFoundation/walletd/pull/120.

**Changes**
- There is now a walletd desktop app.
- The open button now always shows, but is disabled if the daemon is not running.
- The onboarding copy is now more clear about the launcher vs the full user interface.

Screenshot 2024-05-15 at 4 43 39 PM
Screenshot 2024-05-15 at 3 08 55 PM
Screenshot 2024-05-15 at 3 02 20 PM

### CODE REVIEW

**Comment by alexfreska** [comment]
* **#39** <a href="https://app.graphite.dev/github/pr/SiaFoundation/desktop/39?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alexfreska and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: walletd/scripts/download.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+downloadRelease()
+
+async function downloadRelease(): Promise<void> {
+  try {
+    const versionFilePath = path.join(getDaemonDirectoryPath(), 'version')
+    const tag = fs.readFileSync(versionFilePath, { encoding: 'utf8' }).trim()
+
+    const octokit = new Octokit({
+      auth: process.env.GITHUB_TOKEN,
```
Comment (ChrisSchinnerl) [inline walletd/scripts/download.ts:19]: Do we need a token for fetching a release from a public repo?

**Comment by alexfreska** [comment]
Merging this PR and all other approved ones so I can test the Windows build flow 👍🏼 Let me know if there was any further feedback.

**Comment by alexfreska** [comment]
### Merge activity

* **May 17, 3:35 PM EDT**: @alexfreska started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/SiaFoundation/desktop/39).

**Comment by alexfreska** [comment]
Closes https://github.com/SiaFoundation/desktop/issues/13

**Pull request merged**